### ISSUE TITLE
File not found bug

### ISSUE THREAD

**Issue by Fox32** [dart-lang/test#44]
I was trying out the alpha version of unittest 0.12 on Windows 8 and got this output:

```
C:\Users\Oliver\Projects\project-z>pub run unittest:unittest
Failed to load "test\ai_state_machine_test.dart":
Cannot open file, path = 'C:\Users\Oliver\Projects\project-z\test%5Cpackages\uni
ttest\src\runner\vm\isolate_listener.dart' (OS Error: Das System kann den angege
benen Pfad nicht finden.
, errno = 3)
```

Which means "File not found". The file is on disk, but the path is wrong. I'm not sure where the escape sequence %5C is comming from.

**Comment by nex3** [comment]
Can you link a minimal package that demonstrates the issue? Also, which specific version of `unittest` are you using? There are currently two alphas out.

**Comment by Fox32** [comment]
I'm using alpha1. I will create a package later today, but maybe it is related to my package versions. I'm using dependency_overrides ob shelf version 0.6.0 which doesn't work. But changing it to 0.5.7 is working either.

**Comment by Fox32** [comment]
Here is a small package with the right dependencies to reproduce the issue on Windows:
https://dl.dropboxusercontent.com/u/480043/unittest_bug.zip

Steps to reproduce:
1. Run: pub get
2. Run: pub run unittest:unittest

**Comment by nex3** [comment]
@munificent When you have a sec, could you see if you can reproduce this on Windows?

**Comment by munificent** [comment]
In the definition of `packageRootFor()`, change:

```dart
return packageRoot;
```

to:

```dart
return p.toUri(packageRoot).toString();
```

**Comment by nex3** [comment]
Thanks, Bob!

**Comment by seaneagan** [comment]
It is fixed in `unittest` 0.12.0-beta.3, but regressed in 0.12.0-beta.4 for me in windows.

If I depend on `test` 0.12.0-beta.3 I now get a different error where it appears to still be looking for `unittest`:

```
Failed to load "test\invocation_maker_test.dart":
Unhandled exception:
Uncaught Error: FileSystemException: Cannot open file, path = 'C:\Users\se136c\Documents\GitHub\unscripted\master\test\packages\unittest\unittest.dart' (OS Error: The system cannot find the path specified.
, errno = 3)
```

**Comment by nex3** [comment]
@seaneagan Do you think you could dive into it and see if you can figure out what's going on? I don't have a Windows machine and Bob really hates having to use his.
